apps/emqx/include/emqx_persistent_session_internals.hrl

```diff
@@ -68,7 +68,29 @@
     %% marked for deletion:
     unsubscribed = false :: boolean(),
     %% Reference to the subscription state:
-    sub_state_id :: emqx_persistent_session_ds_subs:subscription_state_id()
+    sub_state_id :: emqx_persistent_session_ds_subs:subscription_state_id(),
```

Comment:
We cannot change this record. It already could be stored somewhere.

Can we just store different types of SRS in different containers and track ownership in the code?

Reply:
We probably can, but we cannot treat records polymorphically. All code working with srs should be rewritten to use some polymorphic accessors 🤔

Reply:
We may try to store ownership separately in the session state.

Reply:
Probably we may completely get rid of this

Reply:
In lieu of this remark https://github.com/emqx/emqx/pull/13163#discussion_r1621285076 this should not be needed indeed.